drivers/dma/dma_stm32.c

```diff
@@ -620,11 +620,34 @@ static int dma_stm32_init(struct device *dev)
 	return 0;
 }
 
+static int dma_stm32_get_status(struct device *dev, u32_t id,
+				struct dma_status *stat)
+{
+	const struct dma_stm32_config *config = dev->config_info;
+	DMA_TypeDef *dma = (DMA_TypeDef *)(config->base);
+	struct dma_stm32_data *data = dev->driver_data;
+	struct dma_stm32_stream *stream = &data->streams[id - STREAM_OFFSET];
```

Comment:
stream should be retrieved after bound checks done below

Reply:
Good catch, thank you! I'll change that.

Reply:
Actually, other driver functions seem to do the same. Maybe all should be fixed @erwango ?

Reply:
Current code reflects previous review (https://github.com/zephyrproject-rtos/zephyr/pull/23157#discussion_r391009777).
It would actually be nice to be creative and accomodate both sides so we don't have to come back on this.

Reply:
@erwango the thing is that the resulting `id` needs to be checked before accessing an array element. I think this is a bug that has to be fixed regardless of previous reviews, variable can be declared and initialized after check is successful.

Reply:
Maybe we can create a new bug issue to address this, or include the bugfix together here?

Reply:
@jli157, yes this should be fixed separately

Reply:
Agree, I think that it is good to do it right for the current change though.